### ISSUE TITLE
the `latest` tag should be removed when the CI is done

### ISSUE THREAD

**Issue by 2bndy5** [cpp-linter/cpp-linter-action#84]
Just tried to commit to my backlogged-updates branch:
```
> git pull --tags origin backlogged-updates
From https://github.com/cpp-linter/cpp-linter-action
 * branch            backlogged-updates -> FETCH_HEAD
 ! [rejected]        latest     -> latest  (would clobber existing tag)
```
After #83 was opened, the `latest` tag was moved to a different branch which seemed to cause conflicts when my machine tried to do `git pull`.

I would suggest
```diff
--- a/.github/workflows/run-test.yml
+++ b/.github/workflows/run-test.yml
@@ -23,6 +23,7 @@ jobs:
           git push origin latest
 
   test-action:
+    needs: [add-tag]
     runs-on: ubuntu-latest
     steps:
     - uses: convictional/trigger-workflow-and-wait@v1.6.3
@@ -38,6 +39,7 @@ jobs:
         trigger_workflow: true
         wait_workflow: true
   test-package:
+    needs: [add-tag]
     runs-on: ubuntu-latest
     steps:
     - uses: convictional/trigger-workflow-and-wait@v1.6.3
@@ -52,3 +54,14 @@ jobs:
         propagate_failure: true
         trigger_workflow: true
         wait_workflow: true
+  remove-tag:
+    needs: [test-action, test-package]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: remove latest tag (cleanup)
+        run: |
+          git config user.name 'github-actions'
+          git config user.email '41898282+github-actions[bot]@users.noreply.github.com'    
+          git push --delete origin latest || true
```

**Comment by 2bndy5** [comment]
Removing the tag `latest` let me push/pull again.

**Comment by 2bndy5** [comment]
CI failed with:
```
To https://github.com/cpp-linter/cpp-linter-action
 - [deleted]         latest
To https://github.com/cpp-linter/cpp-linter-action
 ! [remote rejected] latest -> latest (shallow update not allowed)
error: failed to push some refs to 'https://github.com/cpp-linter/cpp-linter-action'
```

**Comment by 2bndy5** [comment]
This is getting annoying when there's 2 PRs open. related to #71

**Comment by shenxianpeng** [comment]
From the CI failed log, tt seems need to checkout all branches and tags.

```yml
- uses: actions/checkout@v3
   with:
     fetch-depth: 0
```

**Comment by shenxianpeng** [comment]
This issue should be fixed in #81, can it be closed?

**Comment by 2bndy5** [comment]
The only thing I did to the workflow in that PR was [add the `needs: [add-tag]` to the subsequent steps](https://github.com/cpp-linter/cpp-linter-action/commit/600b1aabe9753af4fe2c6ad0f4220166505b4252#diff-40b21f1367c9de677ec3d7dda3e3cff777b4eefc23d77eea3f9710e848ae82a6). We can close this for now, but we may encounter this issue again...

I think the main culprit was having more than 1 PR open at the same time, which caused problems moving the `latest` tag around difficult because the repo was checked out with a shallow history
https://github.com/cpp-linter/cpp-linter-action/blob/600b1aabe9753af4fe2c6ad0f4220166505b4252/.github/workflows/run-test.yml#L15-L16
This could use an added
```yml
      with:
        fetch-depth: 0
```
